app/models/ems_event.rb

```diff
@@ -136,6 +137,18 @@ def self.process_host_in_event!(event, options = {})
     end
   end
 
+  def self.process_storage_in_event!(event, options = {})
+    ems_ref = event[:ems_ref]
```

Comment:
`event[:ems_ref]` is not the same thing as the physical_storage_ems_ref.  This is the unique reference _of the event itself_

Prefer `:physical_storage_ems_ref` and either add a column for that also, or delete it from the hash first (first is probably best since we're changing the schema anyway)